### PR TITLE
feat(CSS): add CSS logstash configuration resource

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add CSS logstash configuration resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add CSS logstash configuration resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/css" TESTARGS="-run TestAccLogstashConfiguration_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run TestAccLogstashConfiguration_basic -timeout 360m -parallel 4
=== RUN   TestAccLogstashConfiguration_basic
=== PAUSE TestAccLogstashConfiguration_basic
=== CONT  TestAccLogstashConfiguration_basic
--- PASS: TestAccLogstashConfiguration_basic (880.82s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       880.876s
```

### CODE REVIEW

File: docs/resources/css_logstash_configuration.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+## Import
+
+The CSS logstash configuration can be imported using `cluster_id` and `name` separated by a slash, e.g.
+
+```bash
+$ terraform import huaweicloud_css_logstash_configuration.test <cluster_id>/<name>
+```
+
+Please add the followings if some attributes are missing when importing the resource.
```
Comment (houpeng80) [inline docs/resources/css_logstash_configuration.md]: delete this line

File: docs/resources/css_logstash_configuration.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason.
+The missing attributes include: `sensitive_words`.
+It is generally recommended running `terraform plan` after importing the resource.
+You can then decide if changes should be applied to the instance, or the resource definition should be updated to
+align with the instance. Also you can ignore changes as below.
```
Comment (houpeng80) [inline docs/resources/css_logstash_configuration.md]:
```suggestion
Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
API response, security or some other reason.
The missing attributes include: `sensitive_words`.
It is generally recommended running `terraform plan` after importing CSS logstash configuration.
You can then decide if changes should be applied to CSS logstash configuration, or the resource definition should be updated to align with CSS logstash configuration. Also you can ignore changes as below.
```

File: huaweicloud/provider.go [hunk 3]
```diff
@@ -922,12 +922,13 @@ func Provider() *schema.Provider {
...
-			"huaweicloud_css_snapshot":         css.ResourceCssSnapshot(),
-			"huaweicloud_css_thesaurus":        css.ResourceCssthesaurus(),
-			"huaweicloud_css_configuration":    css.ResourceCssConfiguration(),
-			"huaweicloud_css_scan_task":        css.ResourceScanTask(),
+			"huaweicloud_css_cluster":                css.ResourceCssCluster(),
+			"huaweicloud_css_logstash_cluster":       css.ResourceLogstashCluster(),
+			"huaweicloud_css_snapshot":               css.ResourceCssSnapshot(),
+			"huaweicloud_css_thesaurus":              css.ResourceCssthesaurus(),
+			"huaweicloud_css_configuration":          css.ResourceCssConfiguration(),
+			"huaweicloud_css_logstash_configuration": css.ResourceLogstashConfiguration(),
```
Comment (houpeng80) [inline huaweicloud/provider.go]: move to line 927

File: huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+	_, err = cssV1Client.CreateCnf(&model.CreateCnfRequest{
+		ClusterId: d.Get("cluster_id").(string),
+		Body:      createCnfOpts,
+	})
+	if err != nil {
+		return diag.Errorf("error creating CSS logstash cluster configuration: %s", err)
+	}
+
+	clusterId := d.Get("cluster_id").(string)
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go]: move to line 123, and it should not be get again in line 125

File: huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+		return nil, fmt.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format of import ID, must be <cluster_id>/<name>")
+	}
+
+	d.Set("cluster_id", parts[0])
+	d.Set("name", parts[1])
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go]: deal the error with multierror.Append

File: huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+	_, err = cssV1Client.ShowGetConfDetail(&model.ShowGetConfDetailRequest{
+		ClusterId: d.Get("cluster_id").(string),
+		Name:      d.Get("name").(string),
+	})
+	if err != nil {
+		return []*schema.ResourceData{d}, err
+	}
```
Comment (houpeng80) [inline huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go]: it seems useless, the read method will be called when importing the resouce, this func will be called

File: docs/resources/css_logstash_configuration.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+* `conf_content` - (Required, String) Specifies the configuration file content of the CSS logstash cluster.
+
+* `setting` - (Required, List) Specifies configuration file setting information of the CSS logstash cluster.
+  The [setting](#Css_logstash_configuration_setting) structure is documented below.
+
+* `sensitive_words` - (Optional, List) Specifies the input list of sensitive strings that need to be hidden.
+  After configuring the hidden string list, all strings in the list will be hidden as `***` in the returned
+  configuration content (the list supports a maximum of **20** items, and the maximum length of a single string
+  is 512 bytes).
+  **Note:** When this field is used, the configuration file content will also trigger update changes when the
```
Comment (Jason-Zhang9309) [inline docs/resources/css_logstash_configuration.md]:
```suggestion
  -> **Note:** When this field is used, the configuration file content will also trigger update changes when the
```

File: huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	updateLogstashConfChanges := []string{
+		"conf_content",
+		"setting",
+		"sensitive_words",
+	}
+
+	if d.HasChanges(updateLogstashConfChanges...) {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go]: if there is only one API used in the update func, we don't need to check the changes
```suggestion
```

**Pull request merged**